### DISCUSSION TITLE
2021-09-15 (automatic release)

### DISCUSSION THREAD

**Discussion by pipcet** [pipcet/pearl#23] [General]
https://github.com/pipcet/pearl/releases/tag/2021-09-15

Instructions in the discussion. Please feel free to comment in any way.

**Comment by pipcet** [comment]
This release fixes a number of bugs, and adds a working installer. (Please forgive the optimism both here and in the rest of this post. Not everything might work, and please let me know if it doesn't).

To install pearl, boot into Recovery Mode, select Options, then Terminal from the Utilities menu, then enter

```
curl -L https://github.com/pipcet/pearl/releases/latest/download/pearl.pl|perl
```

Please make sure that you're using HTTPS, as you're feeding a perl script directly into a perl interpreter running as the root user. (It would be a good idea to verify the integrity of the file with `sha512sum`, but that's not available in the Recovery Mode environment. `md5` is, but that's no longer a secure choice.)

After the download completes, you will be prompted for nvram options for pearl. It is a good idea to specify "root /dev/mapper/debian-root" or something similar here, even if that partition does not exist, as it will save you a reboot into Recovery Mode lateron.

After that, make sure you understand what you are doing and, if you decide to go ahead, enter your username and password. This is as good a time as any to remind you there are no warranties and you're doing this at your own risk.

The screen comes back on with a full-screen text console and some text scrolls by.  Do nothing and wait.

The screen comes on again with four terminal windows: the upper left is the menu for interaction, the upper right is a general help/system status window, the lower left is for kernel messages, the lower right is for a shell prompt. There's an uptime counter and a countdown to when Debian will boot automatically. Do nothing and wait.

The screen will freeze for a few seconds and come back on with a full-screen text console. Many messages appear, some of which may sound worrying but are most likely harmless. If you are prompted for a passphrase (multiple times), hit Ctrl-D to skip the prompt; otherwise, do nothing and wait. As the Debian initrd is uncompressed only now, and it uses a comparatively slow compression algorithm, there will be a delay.

If you're on a mini, we now attempt to switch to 4k@60Hz resolution. That might look a bit strange as the framebuffer is switched. The screen goes dark for about ten seconds, then comes back on with a high-resolution text console. Do nothing and wait.

If you've selected a root file system, the installer should start automatically after another delay. Follow its instructions. Otherwise, the `root` user will be logged in automatically and `nmtui`, the NetworkManager interactive curses application, will be started. Select "Activate a connection"; both WiFi networks and the wired ethernet network should appear as appropriate for your device, so you can configure a network connection.

Once you quit `nmtui`, ensure you can reach the internet.

At this point, or later, you can enter "installer" at the shell prompt to launch Debian's graphical installer. In the installer, following the instructions should generally work. In order to boot automatically, create an LVM volume group called `debian` and a volume called `root` (or anything else you specified at the initial prompt; note that the physical volume may be encrypted, and you will be prompted for a decryption password automatically). Root partitions using the btrfs file system require specifying `root /dev/mapper/debian-root -o subvol=@rootfs` or similar at the initial prompt. It is expected that there is a warning about the lack of a known bootloader.

You're now logged in to the `root` account of a live Debian system running from a RAM filesystem: that means any changes you make locally will be lost as soon as you reboot. The RAM file system is initially limited in size. You can increase that by running `mount -o remount,size=7G /`, substituting any reasonable size for "7G".

You can now install, for example, a graphical desktop environment by running:
```
apt-get clean
apt-get update
apt-get install task-gnome-desktop
apt-get clean
init 5
```

To log in, use the non-root user that was created for you: username and password are both "y".

Finally, I'd be happy to hear from you! Please feel free to respond right in this discussion, open an issue, or contact me by email if you have any thoughts, observations, screenshots, problems, comments, or suggestions (or anything else, really).

Technical details:
- both mac addresses are chosen at random rather than set to the physical address.
- on 8 GB machines, there's a long delay as an initramfs is compressed (only to be uncompressed again a few seconds later). I'm working on a better solution for that.
- USB keyboards connected to the mini's Type A slot should work in the final Debian system, but will not work in the Pearl menu. USB keyboards connected to the Type C ports will.

**Comment by Johnnie390** [comment]
Pip, 
thank you for the prompt reply.
I will start testing tomorrow afternoon and/or Friday morning. I am on the road at the moment.
Thank you again for the effort and the expended brain cells...

Regards,

Ry

PS Do I really need to use LVM?, on a small machine like mine, I would prefer to avoid yet another layer of complexity.

**Reply by pipcet** [reply]
> PS Do I really need to use LVM?, on a small machine like mine, I would prefer to avoid yet another layer of complexity.

There's no requirement to use LVM, no. you can specify `root /dev/sda1` or `root /dev/nvme0n1p5` instead, and everything _should_ work fine without LVM, but I haven't tested it (reports welcome). I suppose it's possible to use a uuid path, but that's not very convenient...

**Comment by Johnnie390** [comment]
Pip,

fair enough, will test without LVM.. simplex sigillum veri.

Regards,

Ry
